plugins/inputs/nsd/nsd_test.go

```diff
@@ -0,0 +1,241 @@
+package nsd
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+var TestTimeout = internal.Duration{Duration: time.Second}
+
+func NsdControl(output string, timeout internal.Duration, useSudo bool, server string) func(string, internal.Duration, bool, string) (*bytes.Buffer, error) {
+	return func(string, internal.Duration, bool, string) (*bytes.Buffer, error) {
+		return bytes.NewBuffer([]byte(output)), nil
+	}
+}
+
+func TestParseFullOutput(t *testing.T) {
+	acc := &testutil.Accumulator{}
+	v := &Nsd{
+		run: NsdControl(fullOutput, TestTimeout, true, ""),
+	}
+	err := v.Gather(acc)
+
+	assert.NoError(t, err)
+
+	assert.True(t, acc.HasMeasurement("nsd"))
+
+	hasServerTag := false
+	for _, m := range acc.Metrics {
+		_, ok := m.Tags["server"]
+		if !ok {
+			continue
+		}
+		if m.Measurement == "nsd" {
+			hasServerTag = true
+		}
+	}
+	assert.True(t, hasServerTag)
+
+	assert.Len(t, acc.Metrics, 2)
+	assert.Equal(t, acc.NFields(), 89)
+
+	tags := map[string]string{
+		"server": "0",
+		"source": "localhost",
+		"port":   "8952",
+	}
+	acc.AssertContainsFields(t, "nsd", parsedFullOutput)
+	acc.AssertContainsTaggedFields(t, "nsd", parsedFullOutputServerAsTagMeasurementNsdServers, tags)
```

Comment:
Obviously tests would need updated after changing the measurement name and the above updates.